baybe/surrogates/base.py

```diff
@@ -43,6 +45,11 @@
 
     from baybe.surrogates.composite import CompositeSurrogate
 
+PosteriorStatistic: TypeAlias = float | Literal["mean", "std", "var", "mode"]
+"""Type alias for requestable posterior statistics.
+
+A float will result in the corresponding quantile points."""
```

Comment:
I actually liked your original name better because the definition of the statistic is really independent of the underlying distribution it is calculated from – so why make it more restricted than necessary?
```suggestion
Statistic: TypeAlias = float | Literal["mean", "std", "var", "mode"]
"""Type alias for requestable statistics (a float yields the corresponding quantile)."""
```

Reply:
given thats its only exclusively used and implemented for posterior statistics, its better to be precise
for instance it cant be used for a prior (not in principle, but we have nothing of the sort implemented)

Reply:
Since it's only a type alias, I'm not gonna push it, so if you insist, keep it. Just to highlight: it's another example (from our latest discussion) where we mix the `definition` of an object/quantity with `the intended use by other objects`. To be precise:
* The notion of a `Posterior` does not exist at the level of the statistic, i.e. a `mean` or a `std` is a concept that applies to any distribution
* Yes, right now no of our Priors use it, nor do any other objects. But they easily could, and what would be the consequence it this case? --> we'd need to rename the type alias and strictly speaking create a backwards incompatibility (I know it's no catastrophe for a type alias, but the problem applies likewise to all other objects)

But perhaps you can compress the docstring to a oneliner, as suggested